doc/api/process.markdown

```diff
@@ -183,6 +183,87 @@ this, you can either attach a dummy `.catch(() => { })` handler to
 `resource.loaded`, preventing the `'unhandledRejection'` event from being
 emitted, or you can use the [`'rejectionHandled'`][] event.
 
+## Event: 'warning'
+
+Emitted whenever Node.js emits a process warning.
+
+A process warning is similar to errors in that they describe exceptional
+conditions that are being brought to the user's attention. they are not,
+however, part of the normal Node.js and JavaScript error handling flow. Node.js
+can emit warnings whenever it detects: (a) the use of deprecated APIs, (b) bad
+coding practices that could lead to sub-optimal application performance or bugs.
+
+The event handler for `'warning'` events is called with a single `warning`
+argument whose value is an `Error` object. There are three key properties that
+describe the warning:
+
+* `name` - The name of the warning (currently either `DeprecationWarning`,
+  `BadPracticeWarning`).
+* `message` - A system-provided description of the warning.
+* `stack` - A stack trace to the location in the code where the warning was
+  issued.
+
+```js
+process.on('warning', (warning) => {
+  console.log(warning.name);    // Print the warning name
+  console.log(warning.message); // Print the warning message
+  console.log(warning.stack);   // Print the stack trace
+});
+```
+
+By default, Node.js will print Process Warnings to `stderr`. The `--no-warnings`
+command-line option can be used to suppress the default console output but the
+`'warning'` event will still be emitted by the `process` object.
+
+The following example illustrates the default deprecation warning that is
+printed to `stderr` when a deprecated API is used:
+
+```
+$ node
+> util.debug('foo');
+DEBUG: foo
+undefined
+> DeprecationWarning: (node:94742) util.debug is deprecated. Use
+console.error instead.
+```
+
+In contrast, the following example turns off the default warning output and adds
+a custom handler to the `'warning'` event:
+
+```
+$ node --no-warnings
+> var p = process.on('warning', (warning) => console.log('DEPRECATED API!'));
```

Comment:
Could you use `console.warn()` here, and print the warning instead of the hard coded string?